File: backend/core/manager/ImageUpdater.py
import logging

from core.manager.ImageManager import ImageManager
from core.utils.helper.FilterHelper import FilterHelper
from backend.core.utils.type.data.Phenomena import Phenomena

from typing import List

class ImageUpdater(ImageManager):

    # 更新图片标签
    def updateTags(self, imageId: str, newTags: List[str] | str):
        try:
            tagsToAdd = [newTags] if (isinstance(newTags, str)) else newTags
            idFilter = FilterHelper.idFilter(imageId)
            updateQuery = {
                "$addToSet": {
                    "tags": {
                        "$each": tagsToAdd
                    }
                }
            }
            result = self.db.images.update_one(idFilter, updateQuery)

            return result.modified_count > 0
        except Exception as e:
            logging.error(f"❌ 更新图片标签失败: { str(e) }")

    # 更新图片气象信息
    def updatePhenomena(self, imageId: str, newPhenomena: List[Phenomena] | Phenomena):
        try:
            phenomenaToAdd = [newPhenomena] if (isinstance(newPhenomena, Phenomena)) else newPhenomena
            idFilter = FilterHelper.idFilter(imageId)
            updateQuery = {
                "$addToSet": {
                    "phenomena": {
                        "$each": phenomenaToAdd
                    }
                }
            }
            result = self.db.images.update_one(idFilter, updateQuery)

            return result.modified_count > 0
        except Exception as e:
            logging.error(f"❌ 更新图片气象信息失败: { str(e) }")

    # 更新图片卫星信息
    def updateSatellite(self, imageId: str, newSatellite: str):
        try:
            idFilter = FilterHelper.idFilter(imageId)
            updateQuery = {
                "$set": {
                    "stats.satellite": newSatellite
                }
            }
            result = self.db.images.update_one(idFilter, updateQuery)

            return result.modified_count > 0
        except Exception as e:
            logging.error(f"❌ 更新图片卫星信息失败: { str(e) }")

    # 更新图片传感器信息
    def updateSensor(self, imageId: str, newSensor: str):
        try:
            idFilter = FilterHelper.idFilter(imageId)
            updateQuery = {
                "$set": {
                    "stats.sensor": newSensor
                }
            }
            result = self.db.images.update_one(idFilter, updateQuery)
            
            return result.modified_count > 0
        except Exception as e:
            logging.error(f"❌ 更新图片传感器信息失败: { str(e) }")

File: backend/src/core/services/ImageUpdater.ts
import { FilterHelper } from "../utils/helper/FilterHelper";
import { Phenomena } from "../utils/type/data/Phenomena";
import { ImageManager } from "./ImageManager";

export class ImageUpdater extends ImageManager {
    
    // 更新图像标签
    public async updateTags(imageId: string, newTags: string[] | string): Promise<boolean> {
        try {
            const tagsToAdd = Array.isArray(newTags) ? newTags : [newTags]
            const idFilter = FilterHelper.idFilter(imageId)
            const updateQuery = { $addToSet: { tags: { $each: tagsToAdd } } }
            const result = await this.collection.updateOne(idFilter, updateQuery)

            return result.modifiedCount > 0
        } catch (error) {
            this.logger.error("❌ 更新标签失败:", error)
            throw new Error("更新标签失败")
        }
    }

    // 更新图像气象信息
    public async updatePhenomena(imageId: string, newPhenomena: Phenomena[]): Promise<boolean> {
        try {
            const idFilter = FilterHelper.idFilter(imageId)
            const updateQuery = { $addToSet: { phenomena: { $each: newPhenomena } } }
            const result = await this.collection.updateOne(idFilter, updateQuery)

            return result.modifiedCount > 0
        } catch (error) {
            this.logger.error("❌ 更新气象信息失败:", error)
            throw new Error("更新气象信息失败")
        }
    }

    // 更新图像卫星信息
    public async updateSatellite(imageId: string, newSatellite: string): Promise<boolean> {
        try {
            const idFilter = FilterHelper.idFilter(imageId)
            const updateQuery = { $set: { 'stats.satellite': newSatellite } }
            const result = await this.collection.updateOne(idFilter, updateQuery)

            return result.modifiedCount > 0
        } catch (error) {
            this.logger.error("❌ 更新卫星信息失败:", error)
            throw new Error("更新卫星信息失败")
        }
    }

    // 更新图像传感器信息
    public async updateSensor(imageId: string, newSensor: string): Promise<boolean> {
        try {
            const idFilter = FilterHelper.idFilter(imageId)
            const updateQuery = { $set: { 'stats.sensor': newSensor } }
            const result = await this.collection.updateOne(idFilter, updateQuery)

            return result.modifiedCount > 0
        } catch (error) {
            this.logger.error("❌ 更新传感器信息失败:", error)
            throw new Error("更新传感器信息失败")
        }
    }
}